pkg/controllers/state/cluster.go

```diff
@@ -255,10 +255,12 @@ func (c *Cluster) populateResourceRequests(ctx context.Context, node *v1.Node, n
 	var daemonsetLimits []v1.ResourceList
 	for i := range pods.Items {
 		pod := &pods.Items[i]
+		if podutils.IsTerminal(pod) {
```

Comment:
We've had this bug in so many places now. I wonder if there's a way for us to simply ignore terminal pods as a blanket policy. I think the cluster autoscaler might do this.